library/src/main/java/com/loopj/android/http/AsyncHttpClient.java

```diff
@@ -259,7 +264,11 @@ public void process(final HttpRequest request, final HttpContext context) throws
                 if (authState.getAuthScheme() == null) {
                     AuthScope authScope = new AuthScope(targetHost.getHostName(), targetHost.getPort());
                     Credentials creds = credsProvider.getCredentials(authScope);
-                    if (creds != null) {
```

Comment:
Can we make this if/else to more clever switch, resulting with appropriate AuthScheme assignment based on what is in response? Because I doubt TokenCredentials can be used only in BearerAuthScheme